lib/elixir_bench/github/client_http.ex

```diff
@@ -0,0 +1,50 @@
+defmodule ElixirBench.Github.ClientHTTP do
+  @moduledoc """
+  The Github HTTP client.
+
+  This client uses HTTP requests to gather data from yml files in given
+  repository hosted on Github. It uses the `:hackney` library to actually send
+  the requests and the `yamerl` library to decode the data in the yml format.
+
+  Requests follow a base url:
+
+    - base_url = 'https://raw.githubusercontent.com'
+
+  This module implements the following functions:
+
+    - a `get_yaml/1` function that takes the url path of the project and return
+      the raw parsed data in the yaml format.
+  """
+
+  @behaviour ElixirBench.Github.Client
+
+  def get_yaml(path) do
+    get(url(path), [{"accept", "application/yaml"}], &decode_yaml/1)
+  end
+
+  @doc false
+  defp decode_yaml(content) do
+    case :yamerl.decode(content, [:str_node_as_binary, map_node_format: :map]) do
+      [parsed] -> {:ok, parsed}
+      _ -> {:error, :invalid}
+    end
+  end
+
+  defp get(url, headers, callback) do
+    options = [:with_body, ssl_options: []]
+
+    case :hackney.get(url, headers, <<>>, options) do
```

Comment:
Speaking of erlang and elixir strings. - this is an empty binary and with that we can replace it with an empty string.

```
iex(1)> <<>>
""
iex(2)> ""
""
iex(3)> <<>> = ""
""
iex(4)> <<>> == ""
true
```